lenskit-core/src/main/java/org/lenskit/rerank/NumCandidates.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * LensKit, an open source recommender systems toolkit.
+ * Copyright 2010-2016 LensKit Contributors.  See CONTRIBUTORS.md.
+ * Work on LensKit has been funded by the National Science Foundation under
+ * grants IIS 05-34939, 08-08692, 08-12148, and 10-17697.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package org.lenskit.rerank;
+
+import org.grouplens.grapht.annotation.DefaultDouble;
+import org.lenskit.hybrid.RankBlendingItemRecommender;
+import org.lenskit.inject.Parameter;
+
+import javax.inject.Qualifier;
+import java.lang.annotation.*;
+
+/**
+ * Number of candidate items to get from base recommender. Set to negative for no preference.
```

Comment:
1. Efficiency - This may be pre-mature optimization, but this algorithm is O(n^2) as opposed to (and in addition to) the O(n log (k)) ranking we already do. If we rank over all items it might get expensive so I think we need _some_ system for only taking the best options.
1. Replicate past work - I'm pretty sure past papers have used a constant-n approach. Ziegler used 50 for experiments with top-10 for example. (that said, that could easily be converted to a factor)
2. If we request a factor we should have a parameter for that factor. bringing up back to essentially the same conversation. For lack of a concrete use case I don't want to make it generic between factor and number right now. 
3. Absolute number allows negative numbers covering the "just ask for all" use-case" (so long as the recommender also assumes -1 == all, you left this frustratingly vague in the documentation).

I think the only really general solution is to have that be configureable in a more generic way (to allow "recommend all items above threshold" for example) I don't know if I want that fix to be this PR and I don't know if that part really belongs here or in a more generic level.